[BUG] Iceberg tests fail due to duplication of spark.jarc conf via PYSP_TST and on the command line

**Describe the bug**
In non-xdist pytest we pass spark.jars config twice: via CLI and PYSP_TEST mechanism 

**Steps/Code to reproduce bug**
Run 
```bash
SPARK_HOME=~/dist/spark-3.2.2-bin-hadoop3.2 \
NUM_LOCAL_EXECS=1 \
SPARK_SUBMIT_FLAGS="--packages org.apache.iceberg:iceberg-spark-runtime-3.2_2.12:0.13.2 
  --conf spark.sql.extensions=org.apache.iceberg.spark.extensions.IcebergSparkSessionExtensions 
  --conf spark.sql.catalog.spark_catalog=org.apache.iceberg.spark.SparkSessionCatalog 
  --conf spark.sql.catalog.spark_catalog.type=hadoop 
  --conf spark.sql.catalog.spark_catalog.warehouse=$(mktemp -d -p /tmp spark-warehouse-XXXXXX) 
  --conf spark.rapids.force.caller.classloader=false" \
./integration_tests/run_pyspark_from_build.sh -m iceberg --iceberg \
-k 'test_iceberg_read_partition_key[PERFILE-Long]'
```

And observe:
```
E                   Caused by: java.lang.ClassNotFoundException: org.apache.iceberg.spark.source.SparkWrite$WriterFactory
E                       at java.net.URLClassLoader.findClass(URLClassLoader.java:387)
E                       at java.lang.ClassLoader.loadClass(ClassLoader.java:418)
E                       at java.lang.ClassLoader.loadClass(ClassLoader.java:351)
E                       at java.lang.Class.forName0(Native Method)
E                       at java.lang.Class.forName(Class.java:348)
E                       at org.apache.spark.serializer.JavaDeserializationStream$$anon$1.resolveClass(JavaSerializer.scala:68
```

**Expected behavior**
Iceberg tests should pass

**Environment details (please complete the following information)**
 - Environment location: local bare metal
 - Spark configuration settings related to the issue: spark.jars

**Additional context**
#6044

--- a/integration_tests/run_pyspark_from_build.sh
+++ b/integration_tests/run_pyspark_from_build.sh
@@ -273,10 +273,20 @@ else
             jarOpts=(--driver-class-path "${PYSP_TEST_spark_driver_extraClassPath}")
         fi
 
+        driverJavaOpts="$PYSP_TEST_spark_driver_extraJavaOptions"
+        gpuAllocSize="$PYSP_TEST_spark_rapids_memory_gpu_allocSize"
+
+        # avoid double processing of variables passed to spark in
+        # spark_conf_init
+        unset PYSP_TEST_spark_driver_extraClassPath
+        unset PYSP_TEST_spark_driver_extraJavaOptions
+        unset PYSP_TEST_spark_jars
+        unset PYSP_TEST_spark_rapids_memory_gpu_allocSize
+
         exec "$SPARK_HOME"/bin/spark-submit "${jarOpts[@]}" \
-            --driver-java-options "$PYSP_TEST_spark_driver_extraJavaOptions" \
+            --driver-java-options "$driverJavaOpts" \
             $SPARK_SUBMIT_FLAGS \
-            --conf 'spark.rapids.memory.gpu.allocSize='"$PYSP_TEST_spark_rapids_memory_gpu_allocSize" \
+            --conf 'spark.rapids.memory.gpu.allocSize='"$gpuAllocSize" \
             "${RUN_TESTS_COMMAND[@]}" "${TEST_COMMON_OPTS[@]}"
     fi
 fi

